test/express-route.test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var path = require('path'),
+  helpers = require('yeoman-generator').test,
+  assert = require('yeoman-generator').assert,
+  temp = require('temp').track();
+
+describe('Express Sub Generators Tests', function () {
+  this.timeout(0);
+  /**
+   * Setup the temp directory
+   */
+  before(function (done) {
+    helpers.testDirectory(path.join(__dirname, 'temp'), done);
+  });
+
+  /**
+   * Clean up temp directory
+   */
+  after(function () {
+    temp.cleanup();
+  });
+
+  describe('Generate an express route through the sub-generator', function () {
+    beforeEach(function (done) {
+      helpers.run(path.join(__dirname, '../express-route'))
+        .withOptions({
+          'skip-install': true
+        })
+        .withArguments(['foo'])
+        .withPrompts({
+          'moduleName': 'core'
+        })
+        .on('ready', function (generator) {
+          // this is called right before `generator.run()` is called
+        })
+        .on('end', function () {
+          done();
+        });
+    });
+
+    it('should generate an express route file', function () {
+      assert.file('modules/core/server/routes/foo.server.routes.js');
```

Comment:
@mleanos Well, true, that it can be the case - but I'm slowly moving towards "make route/test/controller etc. name inherited from module name".

Reason is, having multiple sets of logics within one module different enough to have separate routes is a confusing and poor, _very_ poor and confusing design. If someone REALLY wants to hack like that - welcome, MEAN boilerplate supports it. But they would have to do it manually, hacky, and the generator would have the best practicies only in it.